pkgs/by-name/vc/vcpkg/package.nix

```
@@ -14,10 +15,27 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     owner = "microsoft";
     repo = "vcpkg";
     rev = finalAttrs.version;
-    hash = "sha256-eDpMGDtC44eh0elLWV0r1H/WbpVdZ5qMedKh7Ct50Cs=";
+    hash = "sha256-kJrCi0pcm/yhssZqB+am+BJ/HBny9Xf+skJEi3U8bYQ=";
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      VCPKG_BASELINE_COMMIT_SHA=$(git rev-parse --verify HEAD)
+      # We only needed the .git folder for that previous command, so we can safely delete it now.
+      # If we don’t delete it, then we might run into problems (see #8567).
+      rm -r .git
+      # Here's the code that Creates this json file in Visual Studio deployment and the code that reads it:
+      #   https://github.com/microsoft/vcpkg-tool/blob/f098d3e0aaa7e46ea84a1f7079586e1ec5af8ab5/vcpkg-init/mint-standalone-bundle.ps1#L21
+      #   https://github.com/microsoft/vcpkg-tool/blob/f098d3e0aaa7e46ea84a1f7079586e1ec5af8ab5/src/vcpkg/bundlesettings.cpp#L87
+      #
+      # Here's the code that we target with this setting. If we use embeddedsha combined with usegitregistry, vcpkg
+      # will checkout a remote instead of trying to do git operation in the vcpkg root
+      #   https://github.com/microsoft/vcpkg-tool/blob/d272c0d4f5175b26bd56c6109d4c4935b791a157/src/vcpkg/vcpkgpaths.cpp#L920
+      #   https://github.com/microsoft/vcpkg-tool/blob/d272c0d4f5175b26bd56c6109d4c4935b791a157/src/vcpkg/configuration.cpp#L718
+      echo '{ "readonly": true, "usegitregistry": true, "deployment": "Git", "embeddedsha": "'"$VCPKG_BASELINE_COMMIT_SHA"'" }' > vcpkg-bundle.json
+    '';
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ makeWrapper git ];
```

Now that we generate `vcpkg-bundle.json` using a `postFetch` hook, I don’t think that `git` needs to be a native build input anymore:

```suggestion
  nativeBuildInputs = [ makeWrapper ];
```